### PR TITLE
Remove duplicate nft wallet deletion in reorg

### PR DESCRIPTION
**Pull request by ytx1991**
https://github.com/Chia-Network/issue-tracker/issues/119

### CODE REVIEW

**Review by trepca — DISMISSED** [review]

File: tests/wallet/nft_wallet/test_nft_wallet.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ async def test_nft_wallet_creation_and_transfer(two_wallet_nodes: Any, trusted:
         assert False
     await full_node_api.reorg_from_index_to_new_index(ReorgProtocol(uint32(height - 1), uint32(height + 1), ph1, None))
     await time_out_assert(30, get_nft_count, 0, nft_wallet_0)
-    await time_out_assert(30, get_wallet_number, 1, wallet_node_0.wallet_state_manager)
+    await time_out_assert(30, get_wallet_number, 2, wallet_node_0.wallet_state_manager)
```
Comment (trepca) [inline tests/wallet/nft_wallet/test_nft_wallet.py:256]: so we create a new wallet here or why do we have 2 wallets now?

Comment (ytx1991) [reply tests/wallet/nft_wallet/test_nft_wallet.py:256]: The original XCH wallet and DID wallet

File: chia/rpc/wallet_rpc_api.py [hunk 2]
```diff
@@ -264,15 +264,15 @@ async def get_latest_coin_spend(
...
-                    if odd_coin > 1:
-                        raise ValueError("This is not a singleton, multiple children coins found.")
-                if odd_coin == 0:
+                        if odd_coin is not None:
+                            raise ValueError("This is not a singleton, multiple children coins found.")
+                        odd_coin = coin
+                if odd_coin is None:
                     raise ValueError("Cannot find child coin, please wait then retry.")
-                coin_state = coin_state_list[0]
+                coin_state = odd_coin
```
Comment (arvidn) [inline chia/rpc/wallet_rpc_api.py:275]: my reading of this is that the previous code, incorrectly, assumed that the odd coin was always the first one. This fixes it by both ensuring there's only one and returning the correct coin.

is that right?

File: chia/wallet/wallet_state_manager.py [hunk 3]
```diff
@@ -1631,10 +1631,6 @@ async def reorg_rollback(self, height: int) -> List[uint32]:
                 remove: bool = await wallet.rewind(height)
                 if remove:
                     remove_ids.append(wallet_id)
-            if wallet.type() == WalletType.NFT.value:
-                assert isinstance(wallet, NFTWallet)
-                if await wallet.get_nft_count() == 0:
-                    remove_ids.append(wallet_id)
```
Comment (arvidn) [inline chia/wallet/wallet_state_manager.py:1637]: it looks like the test covers this change, but is there a test covering the change in `wallet_rpc_api.py` too?

Comment (ytx1991) [reply chia/wallet/wallet_state_manager.py:1637]: For DID, tests/wallet/did_wallet/test_did.py#test_get_info
For NFT, tests/wallet/nft_wallet/test_nft_wallet.py#test_nft_set_did

Comment (arvidn) [reply chia/wallet/wallet_state_manager.py:1637]: but those tests passed before the fix, right?

Comment (ytx1991) [reply chia/wallet/wallet_state_manager.py:1637]: Covered in the unit test

**Review by arvidn — APPROVED** [review]

**Pull request merged**